### PR TITLE
Install, run and cdn all now honor .nvmrc, removed old bosco-service nodeVersion

### PR DESCRIPTION
**Pull request by cliftonc**
Basically everything just honors the .nvmrc and it uses nvm properly everywhere (nvm use / nvm which).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -38,6 +38,10 @@ Bosco.prototype.init = function(options) {
   self.options.configFile = options.configFile ? path.resolve(options.configFile) : [self.options.configPath, 'bosco.json'].join('/');
   self.options.defaultsConfigFile = [self.options.configPath, 'defaults.json'].join('/');
 
+  // NVM presets
+  self.options.nvmUse = 'source $HOME/.nvm/nvm.sh ; nvm use ; ';
```
Comment (cliftonc) [inline index.js]: We could make these:

```
source $NVM_DIR/nvm.sh ; nvm use ;
```

**Comment by cliftonc** [comment]
Tests not passing seem to be a problem with my mac.

**Comment by geophree** [comment]
The tests for my bosco dir migration to XDG spec have probably never been run on a mac.  I'll take a look today.

File: index.js (hunk 1 above)
Comment (geophree) [inline index.js]: How about `'source ${NVM_DIR:-$HOME/.nvm}/nvm.sh ; nvm use ; '`?  Uses `$NVM_DIR` but defaults to `$HOME/.nvm` if it's not set or empty.

**Comment by geophree** [comment]
Should we add a check to make sure `nvm` is set up properly (e.g. installed in the standard place or where `$NVM_DIR` points to) if we're going to start relying on it for so much?

**Comment by cliftonc** [comment]
Yes, that's a good idea - I'll add a helper function that check it exists, and call it from all the commands that depend on it.

**Comment by cliftonc** [comment]
@geophree the failing tests were me being daft and using the wrong node version vs installed node_modules version - ignore.

File: src/RunWrappers/Node.js [hunk 2]
```diff
@@ -43,23 +43,29 @@ Runner.prototype.listNotRunning = function(detailed, next) {
...
-  var version = semver.parse(service.nodeVersion);
+Runner.prototype.getInterpreter = function(bosco, options, next) {
+  var exec = require('child_process').exec;
+  var interpreter;
+  var found = false;
 
-  if (!version) {
-    return '';
-  }
+  var e = exec(bosco.options.nvmWhich, options.cwd);
```
Comment (geophree) [inline src/RunWrappers/Node.js]: Just use a callback function:
```js
exec(bosco.options.nvmWhich, options.cwd, function(err, stdout, stderr) {
  // ...
});
```

**Comment by geophree** [comment]
Two things.  It looks like my version of `nvm` `0.11.1` doesn't have `nvm which`, so we may need to ensure nvm is at least a certain version.  Also, the current version in github doesn't create a `current` symlink unless you set `$NVM_SYMLINK_CURRENT` to `true`, so we can unset that variable in our subshell, and we won't change anything about the global state when running in a recent `nvm`

**Comment by cliftonc** [comment]
Ok, re. which version, looks like it landed in ```0.21.0```:  https://github.com/creationix/nvm/releases/tag/v0.21.0

**Comment by geophree** [comment]
:+1: This looks good.  I'll separately look into the test failures on mac.

**Comment by cliftonc** [comment]
https://github.com/tes/bosco/releases/tag/0.5.3

**Pull request merged**